lib/collection/src/shards/local_shard.rs

```diff
@@ -492,11 +506,30 @@ impl LocalShard {
                 Err(err) => log::error!("{err}"),
                 Ok(_) => (),
             }
-            bar.inc(1);
+
+            // Update progress bar or show text progress every WAL_LOAD_REPORT_EVERY
+            let progress = i as u64 + 1;
```

Comment:
I think we could rollback to the previous approach using `bar.inc(1)` instead without `enumerate`.
And then within the `if` simply grab `let progress = bar.position`.

Reply:
Done :+1:

Reply:
Perfection.